src/utilities/BatchTableHierarchyExtension.js

```diff
@@ -0,0 +1,199 @@
+import { parseBinArray } from './FeatureTable.js';
+
+function parseProperty( property, buffer, binOffset, length ) {
+
+	if ( ! ( property instanceof Object ) ) {
+
+		throw new Error( 'BatchTableHierarchyExtension: unexpected value found in extension properties.' );
```

Comment:
Lets remove this - I think we have to rely on the input data being well-formed otherwise we will have these kinds of checks everywhere.

Reply:
Agreed, I'll remove that.